src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php

```diff
@@ -271,57 +287,90 @@ public function testClaimAbandonedMessageWithRaceCondition()
     {
         $redis = $this->createMock(\Redis::class);
 
-        $redis->expects($this->exactly(3))->method('xreadgroup')
-            ->willReturnCallback(function (...$args) {
-                static $series = [
-                    // first call for pending messages
-                    [['symfony', 'consumer', ['queue' => '0'], 1, null], []],
-                    // second call because of claimed message (redisid-123)
-                    [['symfony', 'consumer', ['queue' => '0'], 1, null], []],
-                    // third call because of no result (other consumer claimed message redisid-123)
-                    [['symfony', 'consumer', ['queue' => '>'], 1, null], []],
-                ];
-
-                [$expectedArgs, $return] = array_shift($series);
-                $this->assertSame($expectedArgs, $args);
-
-                return $return;
-            })
-        ;
+        // xpending -> xclaim -> xpending -> xreadgroup
+        $redis->expects($this->once())->method('xreadgroup')
+            ->with('symfony', 'consumer', ['queue' => '>'], 1, null) // weren't able to claim the message, asking for new ones
+            ->willReturn([]);
 
-        $redis->expects($this->once())->method('xpending')->willReturn([[
-            0 => 'redisid-123', // message-id
-            1 => 'consumer-2', // consumer-name
-            2 => 3600001, // idle
-        ]]);
+        $redis->expects($this->exactly(2))->method('xpending')->willReturnOnConsecutiveCalls(
```

Comment:
I forgot to replace willReturnOnConsecutiveCalls with the $series trick (see removed code)